### ISSUE TITLE
Rename symbol on a HTML tag inside of JSX globally renames all tags

### ISSUE THREAD

**Issue by egamma** [Microsoft/TypeScript#7458]
_From @xirzec on March 9, 2016 22:24_

1. Open a project that uses TypeScript + JSX (with react.d.ts loaded)
1. Put the cursor over a common HTML tag like `<div>`
1. Hit F2 to rename symbol
1. Rename the div to something else (like span)

Expected: Just the matching open/close is renamed to `<span>`
Actual: Every div in **every** JSX file is now a span and even react.d.ts has had its div attribute in the HTML tag names renamed to span

VS Code version: 0.10.10

_Copied from original issue: Microsoft/vscode#3937_

**Comment by billti** [comment]
This is entirely by design. The element name in a JSX expression is class/interface/function name, and renaming it renames all references to that class/interface/function, just as it would for any other symbol.

If the expectation is that we would only update the open/close tag, then that is a new feature, and not the current `rename` operation.

**Comment by RyanCavanaugh** [comment]
At a minimum we should block the operation when the tag name is sourced in `IntrinsicElements`

**Comment by mhegazy** [comment]
I think we need a general approach here. may not change anything "outside" your project. for instance any symbol from a .d.ts from a node package, or a @typings folder.

**Comment by xirzec** [comment]
Yeah this is a hard problem, I think the expected behavior is a simple tag rename in the JSX markup, but that's only because of how easy it is to fall into expecting HTML language service semantics out of a JSX file.

The reason this was particularly painful in VS Code is that it edits a ton of files with a bad change and Edit -> Undo only reverted it in the currently opened document, leaving it up to the user to manually revert many unexpected edits (hope you committed your changes recently.)

**Comment by Jack-Works** [comment]
is this fixed by https://github.com/microsoft/TypeScript/pull/28066?

**Comment by GMaiolo** [comment]
Is there news about this or a workaround?

**Comment by WilsonNet** [comment]
I just discovered this bug today in the worst way possible.
